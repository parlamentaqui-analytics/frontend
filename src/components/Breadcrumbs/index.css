
.breadcrumb-item a {
    font-weight: lighter;
    font-family: 'Montserrat', sans-serif;
    color: #0059B8;
}

.breadcrumb-item + .breadcrumb-item::before {
    content: ">";
}

.breadcrumb>.active {
    font-weight: normal;
}

.breadcrumb { 
    display: flex;
    flex-wrap: wrap;
    padding: .75rem 1rem;
    margin-bottom: 1rem;
    list-style: none;
    background-color: white;
    border-radius: .25rem;
    font-size: 15px;
};