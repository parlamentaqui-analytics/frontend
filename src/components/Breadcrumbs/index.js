import React from 'react';
import './index.css';
import {
  Breadcrumb, Container
} from 'react-bootstrap';
import { withRouter } from 'react-router-dom';

const Breadcrumbs = (props) => {
  const {
    history,
    location: { pathname }
  } = props;
  const pathnames = pathname.split('/').filter((x) => x);
  return (
    pathnames.length > 0 && (
      <Container className="mb-3 mt-3">
        <Breadcrumb aria-label="breadcrumb">
          <Breadcrumb.Item onClick={() => history.push('/')}>Home</Breadcrumb.Item>
          {pathnames.map((name, index) => {
            const routeTo = `/${pathnames.slice(0, index + 1).join('/')}`;
            const isLast = index === pathnames.length - 1;
            return isLast ? (
              <Breadcrumb.Item active key={name}>{name}</Breadcrumb.Item>
            ) : (
              <Breadcrumb.Item key={name} onClick={() => history.push(routeTo)}>
                {name}
              </Breadcrumb.Item>
            );
          })}
        </Breadcrumb>
      </Container>
    )
  );
};

export default withRouter(Breadcrumbs);
