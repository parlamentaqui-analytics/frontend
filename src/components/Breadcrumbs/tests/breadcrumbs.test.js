import renderer from 'react-test-renderer';
import Breadcrumbs from '..';
import { createMemoryHistory } from 'history'
import { Router } from 'react-router';

test(`breadcrumb is rendered as it should`, () => {
    const history = createMemoryHistory();
    history.push('/deputados')
    const component = renderer.create(
        <Router history={history}>
            <Breadcrumbs />
        </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});