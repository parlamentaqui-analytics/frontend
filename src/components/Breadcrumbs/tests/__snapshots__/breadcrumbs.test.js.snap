// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumb is rendered as it should 1`] = `
<div
  className="mb-3 mt-3 container"
>
  <nav
    aria-label="breadcrumb"
  >
    <ol
      className="breadcrumb"
    >
      <li
        className="breadcrumb-item"
        onClick={[Function]}
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Home
        </a>
      </li>
      <li
        aria-current="page"
        className="breadcrumb-item active"
      >
        deputados
      </li>
    </ol>
  </nav>
</div>
`;
